File: bot-api/src/main/java/gg/mineral/bot/api/math/simulation/PlayerMotionSimulator.java
package gg.mineral.bot.api.math.simulation;

import gg.mineral.bot.api.controls.Keyboard;
import gg.mineral.bot.api.controls.Mouse;
import gg.mineral.bot.api.math.BoundingBox;

public interface PlayerMotionSimulator extends MotionSimulator {

    /**
     * Gets the keyboard.
     *
     * @return the keyboard
     */
    Keyboard getKeyboard();

    /**
     * Gets the mouse.
     *
     * @return the mouse
     */
    Mouse getMouse();

    /**
     * Gets the pitch.
     *
     * @return the pitch
     */
    float getPitch();

    /**
     * Gets the yaw.
     *
     * @return the yaw
     */
    float getYaw();

    /**
     * Gets the bounding box.
     *
     * @return the bounding box
     */
    BoundingBox getBoundingBox();

    /**
     * Resets the player's motion simulator.
     */
    void reset();
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/math/simulation/PlayerMotionSimulator.kt
package gg.mineral.bot.api.math.simulation

import gg.mineral.bot.api.controls.Keyboard
import gg.mineral.bot.api.controls.Mouse
import gg.mineral.bot.api.math.BoundingBox

interface PlayerMotionSimulator : MotionSimulator {
    /**
     * Gets the keyboard.
     *
     * @return the keyboard
     */
    val keyboard: Keyboard

    /**
     * Gets the mouse.
     *
     * @return the mouse
     */
    val mouse: Mouse

    /**
     * Gets the pitch.
     *
     * @return the pitch
     */
    val pitch: Float

    /**
     * Gets the yaw.
     *
     * @return the yaw
     */
    val yaw: Float

    /**
     * Gets the bounding box.
     *
     * @return the bounding box
     */
    val boundingBox: BoundingBox

    /**
     * Resets the player's motion simulator.
     */
    fun reset()
}
